spec/requests/api/status_spec.rb

```diff
@@ -1,185 +1,218 @@
-      it "returns both" do
-        create(:project, platform: "NPM", name: "bcrypt")
-        create(:project, platform: "Pypi", name: "bcrypt")
+      it "correctly handles go redirects" do
+        project = create(:project, platform: "Go", name: "known/project")
+        requested_name = "unknown/project"
+        allow(PackageManager::Go)
+          .to receive(:project_find_names)
+          .with(requested_name)
+          .and_return([project.name])
 
         post(
-          "/api/check",
+          url,
           params: {
             api_key: internal_user.api_key,
             projects: [
-              { name: "bcrypt", platform: "npm" },
-              { name: "bcrypt", platform: "pypi" },
-            ]
+              { name: requested_name, platform: project.platform },
+            ],
+            score: true,
           }
         )
 
         expect(response).to have_http_status(:success)
-        json = JSON.parse(response.body)
-        expect(json.size).to eq(2)
-        expect(json.select { |p| p[:name] == "bcrypt" && p[:platform] == "NPM" }).to be
-        expect(json.select { |p| p[:name] == "bcrypt" && p[:platform] == "Pypi" }).to be
+        expect(JSON.parse(response.body).dig(0, "name")).to eq(requested_name)
+        expect(JSON.parse(response.body).dig(0, "canonical_name")).to eq(project.name)
+      end
+
+      context "with normal API key" do
+        it "returns no maintenance stats or updated_at" do
+          post url, params: { api_key: normal_user.api_key, projects: [{ name: project_django.name, platform: project_django.platform }] }
+          expect(response).to have_http_status(:success)
+          expect(response.content_type).to eq("application/json")
+          expect(response.body.include?(project_django.name)).to be == true
+
+          json_response = JSON.parse(response.body)
+          expect(json_response.first.key?("repository_maintenance_stats")).to be false
+          expect(json_response.first.key?("updated_at")).to be false
+        end
+      end
+
+      context "with two projects that have the same name but different platforms" do
+        it "returns both" do
+          create(:project, platform: "NPM", name: "bcrypt")
+          create(:project, platform: "Pypi", name: "bcrypt")
+
+          post(
+            url,
+            params: {
+              api_key: internal_user.api_key,
+              projects: [
+                { name: "bcrypt", platform: "npm" },
+                { name: "bcrypt", platform: "pypi" },
+              ],
+            }
+          )
+
+          expect(response).to have_http_status(:success)
+          json = JSON.parse(response.body)
+          expect(json.size).to eq(2)
+          expect(json.select { |p| p[:name] == "bcrypt" && p[:platform] == "NPM" }).to be
+          expect(json.select { |p| p[:name] == "bcrypt" && p[:platform] == "Pypi" }).to be
+        end
       end
     end
   end
+
+  xit "is performant with a bunch of projects" do
```

Comment:
Leaving this here but not running normally. I expect to iterate a little bit here in the future and would like to continue to use this.